xlators/features/locks/src/common.c

```diff
@@ -854,7 +854,6 @@ __insert_and_merge(pl_inode_t *pl_inode, posix_lock_t *lock)
 
                 __destroy_lock(lock);
                 INIT_LIST_HEAD(&sum->list);
```

Comment:
You can remove this INIT_LIST_HEAD too. Even that is happening inside add_locks()